### PR TITLE
Add logging in with QR code

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-14110
affects: student
release note: Added the ability to log in with a QR code

Test plan:
- Log into web at mobiledev.beta.instructure.com > hodor (ask me for password)
- Global Nav > Account > QR for Mobile Login
- Launch the Student App and log out or change users
- Tap 'QR Login'
- Tap Next on Tutorial
- Scan the QR code from web
- Should get logged in
- Scan a QR code that isn't a login code
- Should get an error

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/jzqd5cx) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fjzqd5cx)
teacher: [Link](https://rebrand.ly/mgpmudr) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fmgpmudr)
parent: [Link](https://rebrand.ly/t4zzm0t) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Ft4zzm0t)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Added the ability to log in with a QR code

#### Affected Apps: Student

[MBL-14110](https://instructure.atlassian.net/browse/MBL-14110)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Core/Core/Views/ScannerViewController.swift:104:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    Linting 'PageDetailsPresenter.swift' (669/719)
    Linting 'APINotificationPreferences.swift' (670/719)
</details>
<details><summary>:x: XCTest failed: CoreTests/PlannerListViewControllerTests/testLayout</summary>


    Assertion Failure: PlannerListViewControllerTests.swift:83: XCTAssertEqual failed: ("nil") is not equal to ("Optional("note")")
    Assertion Failure: PlannerListViewControllerTests.swift:85: XCTAssertEqual failed: ("nil") is not equal to ("Optional(true)")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Views/ScannerViewController.swift [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        if (captureSession?.isRunning == true) {
+            captureSession.stopRunning()
+        }
+    }
+
+    func metadataOutput(_ output: AVCaptureMetadataOutput, didOutput metadataObjects: [AVMetadataObject], from connection: AVCaptureConnection) {
+        captureSession.stopRunning()
```
Comment (vanwagonet) [inline Core/Core/Views/ScannerViewController.swift]: do we want to stopRunning before we've been successful? There are many cases in this function that would leave this as just a black screen, right?

File: Core/Core/Views/ScannerViewController.swift [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+        // bottom left
+        path.move(to: CGPoint(x: curveLength, y: rect.height))
+        path.addQuadCurve(to: CGPoint(x: 0, y: rect.height - curveLength), controlPoint: CGPoint(x: 0, y: rect.height))
+
+        arc.path = path.cgPath
+    }
+
+    func updateSize() {
+//        arc.frame = bounds
```
Comment (vanwagonet) [inline Core/Core/Views/ScannerViewController.swift]: do we still need this func? Or did implementing draw make this unnecessary?

File: Core/Core/Views/ScannerViewController.swift (hunk 1 above)
Comment (armstrongnate) [reply Core/Core/Views/ScannerViewController.swift]: You are right, I think only doing this on success would be best. it doesn't make the screen black but it does freeze the frame.

File: Core/Core/Views/ScannerViewController.swift (hunk 2 above)
Comment (armstrongnate) [reply Core/Core/Views/ScannerViewController.swift]: Ah left this here by mistake. Thanks!

File: Core/Core/Views/ScannerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        prompt.pin(inside: promptContainer, leading: 16, trailing: 16, top: 16, bottom: 16)
+
+        let cancel = UIButton()
+        cancel.translatesAutoresizingMaskIntoConstraints = false
+        cancel.backgroundColor = UIColor.named(.backgroundDarkest).withAlphaComponent(0.9)
+        cancel.setImage(UIImage.icon(.x), for: .normal)
+        cancel.tintColor = .named(.textLightest)
+        cancel.addTarget(self, action: #selector(cancelTapped(_:)), for: .primaryActionTriggered)
+        cancel.layer.cornerRadius = 25
+
```
Comment (inst-danger) [inline Core/Core/Views/ScannerViewController.swift]:
```suggestion

```

**Comment by cobbal** [comment]
This doesn't seem to be working in the teacher app.

**Comment by armstrongnate** [comment]
@cobbal good call, I forgot to test Teacher.

Looks like we're going to hold off on Teacher right now because it will require some back-end work. I'll make that change.

**Pull request merged**